DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -325,154 +257,46 @@ function Test-TargetResource
-    }
-
-    if($DisplayGroup)
-    {
-        $parameters["Group"] = $DisplayGroup
-    }
-    else
-    {
-        $parameters["Group"] = $DefaultGroup
-    }
-
-    if($State)
-    {
-        if($State -eq "Enabled")
-        {
-            $parameters["Enabled"] = "True"
-        }
-        else
-        {
-            $parameters["Enabled"] = "False"
-        }
-    }
-
-    if($Profile)
-    {
-        $parameters["Profile"] = $Profile
-    }
-
-    if($Direction)
-    {
-        $parameters["Direction"] = $Direction
-    }
-
-    if($Access)
-    {
-        $parameters["Action"] = $Access
-    }
-
-    if($RemotePort)
-    {
-        $parameters["RemotePort"] = $RemotePort
-    }
-
-    if($LocalPort)
-    {
-        $parameters["LocalPort"] = $LocalPort
-    }
-
-    if($Protocol)
-    {
-        $parameters["Protocol"] = $Protocol
-    }
-
-    if($Description)
-    {
-        $parameters["Description"] = $Description
-    }
-
-    if($ApplicationPath)
-    {
-        $parameters["Program"] = $ApplicationPath
-    }
-
-    if($Service)
-    {
-        $parameters["Service"] = $Service
-    }
-
-    Write-Verbose "SET: Invoke Set-NetFirewallRule [$Name] with splatting its parameters"
-    & $commandName @parameters
-}
-
 # Function to validate if the supplied Rule adheres to all parameters set
-function Test-RuleHasProperties
+function Test-RuleProperties
 {
     param (
         [Parameter(Mandatory)]
         $FirewallRule,
-        [String]$Name,
-        [String]$DisplayGroup,
-        [String]$State,
-        [String[]]$Profile,
-        [String]$Direction,
-        [String]$Access,
-        [String[]]$RemotePort,
-        [String[]]$LocalPort,
-        [String]$Protocol,
-        [String]$Description,
-        [String]$ApplicationPath,
-        [String]$Service
+        [String] $Name,
+        [String] $DisplayName = $Name,
+        [String] $DisplayGroup,
+        [String] $Group,
+        [String] $Enabled,
+        [String[]] $Profile,
+        [String] $Direction,
+        [String[]] $RemotePort,
+        [String[]] $LocalPort,
+        [String] $Protocol,
+        [String] $Description,
+        [String] $ApplicationPath,
+        [String] $Service
     )
 
-    $properties = Get-FirewallRuleProperty -FirewallRule $FirewallRule -Property All
+    $properties = Get-FirewallRuleProperty -FirewallRule $FirewallRule
 
     $desiredConfigurationMatch = $true
 
     if ($Name -and ($FirewallRule.Name -ne $Name))
     {
-        Write-Verbose "Test-RuleHasProperties: Name property value - $FirewallRule.Name does not match desired state - $Name"
-
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($Access -and ($FirewallRule.Action -ne $Access))
-    {
-        Write-Verbose "Test-RuleHasProperties: Access property value - $($FirewallRule.Action) does not match desired state - $Access"
-
+        Write-Verbose "$($MyInvocation.MyCommand): Name property value - $FirewallRule.Name does not match desired state - $Name"
```

Comment:
I think $FirewallRule.Name needs to be enclosed in parentheses - e.g. $($FirewallRule.Name)